crates/matrix-sdk/src/sliding_sync/mod.rs

```diff
@@ -621,6 +631,32 @@ impl SlidingSync {
     pub fn stop_sync(&self) -> Result<()> {
         Ok(self.inner.internal_channel_send(SlidingSyncInternalMessage::SyncLoopStop)?)
     }
+
+    /// Force reloading the to-device token, if caching was enabled for this
+    /// instance of sliding sync.
+    ///
+    /// This should only be done if multiple processes may be reading/writing
+    /// into the sliding sync cache.
+    pub async fn reload_to_device_token(&self) -> Result<()> {
```

Comment:
Hmm, instead of this, perhaps we can just remove the cache for the token?

Reply:
Oh what I meant is that we remove the in-memory cache, and instead always use the store to access the token.

You are correct, the token is crucial so we don't end up consuming the same to-device events.

Reply:
Maybe we should update the comment for this method, to provide the info of this discussion :-).

Reply:
Hmm if sliding sync caching isn't enabled, then there's no transient place where to store the to-device token :/ I'll get rid of it in a follow-up too.